views/js/runner/plugins/tools/answerMasking/plugin.js

```diff
@@ -0,0 +1,229 @@
+/*
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2017 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ */
+/**
+ * @author Christophe Noël <christophe@taotesting.com>
+ */
+define([
+    'jquery',
+    'lodash',
+    'i18n',
+    'taoTests/runner/plugin',
+    'ui/hider',
+    'util/shortcut',
+    'util/namespace',
+    'taoQtiTest/runner/plugins/tools/answerMasking/answerMasking'
+], function ($, _, __, pluginFactory, hider, shortcut, namespaceHelper, answerMaskingFactory) {
+    'use strict';
+
+    /**
+     * The public name of the plugin
+     * @type {String}
+     */
+    var pluginName = 'answerMasking';
+
+    /**
+     * The prefix of actions triggered through the event loop
+     * @type {String}
+     */
+    var actionPrefix = 'tool-' + pluginName + '-';
+
+    /**
+     * Stores the masking state for each item in the test
+     * @type {Object}
+     */
+    var itemStates = {};
+
+    /**
+     * Default Configuration
+     */
+    var defaultConfig = {
+        restoreStateOnToggle: true,
+        restoreStateOnMove: true
+    };
+
+    /**
+     * Returns the configured plugin
+     */
+    return pluginFactory({
+
+        name: pluginName,
+
+        /**
+         * Initialize the plugin (called during runner's init)
+         */
+        init: function init() {
+            var self = this;
+
+            var testRunner = this.getTestRunner(),
+                testData = testRunner.getTestData() || {},
+                testConfig = testData.config || {},
+                pluginConfig = _.defaults((testConfig.plugins || {})['answer-masking'] || {}, defaultConfig),
```

Comment:
Even if this naming is correct, it should be better to use the exact plugin name for more consistency.